src/models/StyledComponent.js

```diff
@@ -235,20 +234,22 @@ export default (ComponentStyle: Function, constructWithOptions: Function) => {
             ? rules
             : rulesFromOptions.concat(rules)
 
-        const newComponentId =
-          previousComponentId &&
-          `${previousComponentId}-${ComponentStyle.generateName(
-            newRules.join(''),
-          )}`
-
         const newOptions = {
           ...optionsToCopy,
-          componentId: newComponentId,
           rules: newRules,
           ParentComponent: StyledComponent,
         }
 
-        return constructWithOptions(createStyledComponent, target, newOptions)
+        const createWithHash = (tag, opts, css) => {
```

Comment:
@neoziro In this case we might want to think about a new option instead. Something like `parentId` which just passes down the old `componentId`. Then we let `componentId` be `undefined` and inside the StyledComponent factory we pass the `parentId` into `generateId` and prepend it to the new id.

cc @geelen